### PR TITLE
OSSMDOC-705: Update content  for Service Mesh Extension API deprecation

### PR DESCRIPTION
**Pull request by rh-tokeefe**
Version(s): 4.9+

Issue: https://issues.redhat.com/browse/OSSMDOC-705

Link to docs preview: http://file.bos.redhat.com/tokeefe/OSSMDOC-705/service_mesh/v2x/ossm-extensions.html#ossm-extensions-migration-overview_ossm-extensions

QE review:
- [x] QE has approved this change.

Additional information:

Reviews complete:
dev: John Wendell
qe: Praneeth Wajjuri
peer: Rolfe

### CODE REVIEW

**Review by jwendell — APPROVED** [review]

**Comment by jwendell** [comment]
You can also remove the note
```
Because both plug-ins will be called for every request, you might want to remove your existing ServiceMeshExtension resource before creating the new WasmPlugin resource. You might get undesired results having two plug-ins active at the same time.
```

**Review by jwendell — APPROVED** [review]

**Review by pbajjuri20 — APPROVED** [review]

File: modules/ossm-extensions-migration-overview.adoc [hunk 1]
```diff
@@ -6,7 +6,7 @@ This module included in the following assemblies:
 [id="ossm-extensions-migration-overview_{context}"]
 = Migrating from `ServiceMeshExtension` to `WasmPlugin` resources
 
-The `ServiceMeshExtension` API is deprecated as of {SMProductName} version 2.2 and will be removed in a future release. If you are using the `ServiceMeshExtention` API, you must migrate to the `WasmPlugin` API to continue using your WebAssembly extensions.
+The `ServiceMeshExtension` API is deprecated as of {SMProductName} version 2.2 and was removed as of {SMProductName} version 2.3. If you are using the `ServiceMeshExtention` API, you must migrate to the `WasmPlugin` API to continue using your WebAssembly extensions.
```
Comment (rolfedh) [inline modules/ossm-extensions-migration-overview.adoc:9]:
```suggestion
The `ServiceMeshExtension` API, which was deprecated in {SMProductName} version 2.2, is removed in {SMProductName} version 2.3. If you are using the `ServiceMeshExtention` API, you must migrate to the `WasmPlugin` API to continue using your WebAssembly extensions.
```
Trying to follow the present tense style we've used in https://redhat-documentation.github.io/supplementary-style-guide/#release-notes.

Comment (rolfedh) [reply modules/ossm-extensions-migration-overview.adoc:9]: Reviewing the openshift-docs repo, there's a terminology split:
- 36 references to WebAssembly module
- 15 references to WebAssembly extension

File: modules/ossm-extensions-overview.adoc [hunk 2]
```diff
@@ -16,7 +16,7 @@ WebAssembly modules can be run on many platforms, including proxies, and have br
 
 [NOTE]
 ====
-When creating new WebAssembly extensions, use the WasmPlugin API. The ServiceMeshExtension API is deprecated as of {SMProductName} version 2.2 and will be removed in a future release.
+When creating new WebAssembly extensions, use the WasmPlugin API. The ServiceMeshExtension API is deprecated as of {SMProductName} version 2.2 and was removed as of {SMProductName} version 2.3.
```
Comment (rolfedh) [inline modules/ossm-extensions-overview.adoc:19]:
```suggestion
When creating new WebAssembly extensions, use the `WasmPlugin` API. The `ServiceMeshExtension` API, which was deprecated in {SMProductName} version 2.2, is removed in {SMProductName} version 2.3.
```

Comment (rolfedh) [reply modules/ossm-extensions-overview.adoc:19]: Similar for the following instances.

**Review by rolfedh — APPROVED** [review]
Hi, Tim. Nice work. Some of the API objects mentioned below need to be placed between backticks. The other wordsmithing is completely optional. I've tried to align the wording with the language in our release note templates.

**Comment by rh-tokeefe** [comment]
@jwendell I've completed a global search and updated all instances of that typo. Thanks again for the attention to detail!

**Review by rolfedh — APPROVED** [review]
Looks ready to merge!

**Pull request merged**